### DISCUSSION TITLE
Push notifications help

### DISCUSSION THREAD

**Discussion by justasxz** [dotnet/maui#6557] [Q&A] [answered]
Hello guys, sorry to bother you with such newbie questions, but I have been struggling to implement push notifications for the last 20+ hours, it's my first time trying to implement push notifications, so I don't know much at all. Basically I tried to follow online tutorials for xamarin, but it's been hell, third party plugins don't work at all, and the basic implementation doesn't work properly. Last attempt was with Microsoft official azure notification hub tutorial. This is the link I followed "https://docs.microsoft.com/en-us/azure/notification-hubs/xamarin-notification-hubs-push-notifications-android-gcm". Besides that could someone recommend and tell me how to set up working push notifications. I created a firebase project downloaded google.json set it's build action, and after that I just been struggling with every different tutorial, main issues I faced are: 

1. Almost all tutorials use a function OnCreate inside MainActivity, mine has no such functions, it's basically empty. I created my own OnCreate, but I am not sure is this even correct approach. 
2. Every tutorial has one missing property or value, that can't be received. 
3. Android manifest file, seems to be different in every tutorial, could someone advice ? 
4. Even adding <application> </application> for the intents and receivers, breaks the app, the app doesn't launch with <application> tag. in manifest. It asks to add debugable propperty if I do add it it just says deploy completed but never actually starts in the device.

**Comment by justasxz** [comment, marked as answer]
The tutorial is a little outdated I figured out that I needed to add muttable or immutable flag to my pendingintent.getactivity, after adding that it works like a charm.

**Reply by jzhao8402** [reply]
Hi justasxz, do you mind to outline how you implement the firebase push notification? I never use Xamarin Forms before but Xamarin Android. I'm looking for how to do it in MAUI.

**Reply by overtongeist** [reply]
hey @justasxz did you use PushSharp for the server side?

**Reply by SergTomcat** [reply]
Thank you. Flags did the trick, azure working for now.

**Reply by Ghevi** [reply]
@jzhao8402 do you still need help with that?

**Reply by jzhao8402** [reply]
@Ghevi Yes, please. We started the migration to MAUI earlier this year but had to stop because of the slowness and features missing of MAUI. Hope MAUI now is stable enough and has more supported common features,  and firebase push notification is one of the must haves.

**Reply by JustasKvederis** [reply]
@jzhao8402  I know it's not the answer you need for now, but if you check out the link in the issue description. I followed it to implement the notifications: https://learn.microsoft.com/en-us/azure/notification-hubs/xamarin-notification-hubs-push-notifications-android-gcm

It requires azure hub, but it does work, just don't forget the immutable flag as mentioned in the answer.

**Reply by jzhao8402** [reply]
@Ghevi I'll try it, thanks.
